File: SumTwo/SumTwo.cpp
// SumTwo.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include <vector>
#include <map>

using namespace std;

vector<int> twoSum(vector<int> &numbers, int target)
{
    //Key is the number and value is its index in the vector.
	map<int, int> hash;
	vector<int> result;
	for (int i = 0; i < numbers.size(); i++) {
		int numberToFind = target - numbers[i];

            //if numberToFind is found in map, return them
		if (hash.find(numberToFind) != hash.end()) {
                    //+1 because indices are NOT zero based
			result.push_back(hash[numberToFind] + 1);
			result.push_back(i + 1);			
			return result;
		}

            //number was not found. Put it in the map.
		hash[numbers[i]] = i;
	}
	return result;
}

int main(int argc, char* argv[])
{
	int iarray[] = {1,4,6,8};
	vector<int> vect(iarray, iarray+4);
	vector<int> retvect;
	retvect = twoSum(vect, 10);

	printf("retvect[%d]\tretvect[%d]\n", retvect[0], retvect[1]);
	printf("Hello World!\n");
	return 0;
}

File: SumTwo/SumTwo.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums) == 0:
            return None
            
        _nums=[]
        for i in nums:
            numberToFind = target-i
            nums_l = len(nums)-1 
            """防止下标越界"""
                           
            if numberToFind in nums:
                if numberToFind == i and nums.count(i) >= 2:
                    try:
                        sub1 = nums.index(i)
                        sub2 = nums[sub1+1:].index(numberToFind)+sub1+1
                    except IndexError as e:
                        print("The index Error: ", e)
                        raise
                else:
                    try:
                        sub1 = nums.index(i)
                        sub2 = nums.index(numberToFind)
                    except IndexError as e:
                        print("The index Error: ", e)
                        raise
                
                if sub1 != sub2:
                    _nums.append(sub1)
                    _nums.append(sub2)
                    return _nums
